### DISCUSSION TITLE
Vanilla Javascript Routing

### DISCUSSION THREAD

**Discussion by ddowy** [vikejs/vike#675] [Discussion]
Does vite-plugin-ssr support just plain vanilla js and html? I am trying to set up a router for my front end that would route paths like /contact to contact.html and /about to about.html then just make async calls to my backend. Can I achieve this with this vite plugin? If not could someone point me in the right direction to achieve this? Thank you.

**Comment by brillout** [comment]
> Does vite-plugin-ssr support just plain vanilla js and html? 

Yes

**Reply by ddowy** [reply]
Is there a page on your website that goes over how to use .html files with your plugin? Or does it have to be rendered from a .page.server.js file?

**Reply by brillout** [reply]
I mean HTML itself: VPS doesn't support `.html` files (for good reasons).
